api/ProfileName/[profileAddress].ts

```diff
@@ -0,0 +1,59 @@
+import assert from 'assert';
+
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+import { isAddress } from 'ethers/lib/utils';
+
+import { adminClient } from '../../api-lib/gql/adminClient';
+import { errorResponse, NotFoundError } from '../../api-lib/HttpError';
+
+const CACHE_SECONDS = 60 * 5;
+export default async function handler(req: VercelRequest, res: VercelResponse) {
```

Comment:
this handler doesn't have any permissions on it, and so this will expose ALL user names for all addresses in our DB.

I think we should just use a normal graphql query to request this data and use the logged in users auth token to authenticate the request. Is there any reason this should be a serverless function?

Reply:
the name already exposed in getCosoulData from api/cosoul/[address] so it's not a sensitive data anymore and I didn't want to change the permission of the column

Reply:
I will replace it with hasura action

Reply:
I will replace it with hasura action

Reply:
> I will replace it with hasura action
For type-safety and permission

Reply:
I don't think we need an action - just use the normal graphql interface exposed by hasura, right @sshmm

Reply:
@crabsinger thank you. I removed the action and used a query instead